packages/stylist-core/Cargo.toml

```diff
@@ -0,0 +1,53 @@
+[package]
+name = "stylist-core"
+version = "0.7.0"
+license = "MIT"
+repository = "https://github.com/futursolo/stylist-rs"
+authors = [
+    "Kaede Hoshiakwa <futursolo@icloud.com>",
+    "Martin Molzer <ga65guy@mytum.de>",
+]
+edition = "2018"
+description = "Stylist is a CSS-in-Rust styling solution for WebAssembly Applications."
+keywords = [
+    "CSS",
+    "web",
+    "css-in-rust",
+    "yew"
+]
+categories = ["wasm", "web-programming"]
+readme = "README.md"
+homepage = "https://github.com/futursolo/stylist-rs"
+
+[dependencies]
+once_cell = "1.8.0"
+yew = { version = "0.18.0", optional = true, default-features = false, features = ["web_sys"] }
+rand = { version = "0.8.4", features = ["small_rng"] }
+
+[dev-dependencies]
+log = "0.4"
+env_logger = "0.9"
+wasm-bindgen-test = "0.3.0"
+
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+getrandom = { version = "0.2.3", features = ["js"] }
+wasm-bindgen = { version = "^0.2.74" }
+web-sys = { version = "0.3.51", features = [
+    "Window",
+    "Document",
+    "Element",
+    "HtmlElement",
+    "HtmlHeadElement",
+    "HtmlStyleElement",
+] }
+
+[target.'cfg(target_arch = "wasm32")'.dev-dependencies]
+wasm-bindgen-test = "0.3.24"
+
+[features]
+default = []
+yew_integration = ["yew"]
```

Comment:
The feature name has been changed back to `yew_integration`, as having a feature and optional dependency of the same name is [currently not possible](https://github.com/rust-lang/cargo/issues/5565). Renaming the dependency internally to something like `yew_dep = { version = ..., package = "yew" }` messes with some `yew` macros, so  I didn't bother going that route.

Reply:
If my understanding is correct, it's possible as long as you are not relying on additional optional dependencies.
You can try one of the examples in the examples folder on the master branch.

> Optional dependencies implicitly define a feature of the same name as the dependency. This means that the same cfg(feature = "gif") syntax can be used in the code, and the dependency can be enabled just like a feature such as --features gif (see Command-line feature options below).

https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

Reply:
Yeah, this is complicated by the fact that `stylist` itself does not (directly) depend on yew, but `stylist-core` does. So the feature flag in `stylist` should trigger the one in `stylist-core`. But alas, since `stylist` has doc-tests that depend on `yew`, it has a `dev-dependency` on `yew`, which is additionally not optional!

Reply:
<img width="787" alt="image" src="https://user-images.githubusercontent.com/11693215/128647753-e0a16fc5-0cc5-4038-99ac-424dd5ea69cc.png">

I don't think that you need to list it as a dev-dependency. But if you want to change it to `yew-integration` it's fine with me as well.

Reply:
On master it works fine! With the proposed changes it would not. Can you make sure you have checked out this pull request when testing?

```
git fetch origin refs/pull/10/head
git checkout FETCH_HEAD
```

Here it shouldn't work when you remove yew from the dev-dependencies (since it also isn't in dependencies)

Reply:
I see your point. Sorry about that.